components/econet/econet.cpp

```diff
@@ -603,9 +611,41 @@ void Econet::homeassistant_read(std::string datapoint_id) {
     }
     capi_.fire_homeassistant_event("esphome.econet_event", data);
   });
-  datapoint_ids_for_read_service_.push(datapoint_id);
+  datapoint_ids_for_read_service_.push(std::pair<std::string, uint32_t>(datapoint_id, src_adr_));
 }
 
+// Called from a Home Assistant exposed service to read a datapoint.
+// Fires a Home Assistant event: "esphome.econet_event" with the response.
+void Econet::homeassistant_read_from_address(std::string datapoint_id, uint32_t address) {
```

Comment:
This is almost identical to existing homeassistant_read. Can't you default the address to src_adr_? Or default it to 0 and use src_adr_ if it's 0.

Reply:
We can't default it, it doesn't appear that there are optional service parameters. We could require/allow someone to enter a 0 to mean "direct towards current value of src_adr_".

It would be a breaking change though.

Reply:
https://github.com/esphome/issues/issues/4996

Reply:
Ah yes I had the same issue for the write service so I ended up creating two write variants. What you have is fine. Change the function above to call this with src_adr_ to avoid all the duplicate code.

Reply:
Or better just have a single homeassistant_read function with the address defaulting to 0. And from yaml call:
`id(econet_id).homeassistant_read(datapoint_id);`
and
`id(econet_id).homeassistant_read_from_address(datapoint_id, address);`

Reply:
Good idea. Done.